### DISCUSSION TITLE
What are the plans for MS Word support going forward?

### DISCUSSION THREAD

**Discussion by half-normal** [quarto-dev/quarto-cli#6114] [Q&A]
### Description

👋 😄 

**Context:**

I have recently moved from RMarkdown to Quarto markdown to produce reports at work (mostly scientific/data analysis reports). Some clients insist that reports are delivered to them in Docx format and my company has fairly strict guidelines for report styling. Majority of the styles can be incorporated using the docx template as instructed in the Quarto documentation. However there are a couple things which currently limit my report writing for Docx:

* Lists of tables and figures - As far as I can tell quarto does not offer much support for these.  I often need to have separate lists for the main report and appendices. This was previously possible in RMarkdown with the Officedown package.

* Cover pages - My company has a set cover section (i.e. multiple pages) which uses roman numerals as page numbers (whereas the main text uses regular number) and includes things like the authors, date, title, etc. Some of these attributes are repeated in the main section of the document.

* Yaml attributes - It would be nice to be able to access the yaml attributes (e.g. title, authors etc) within the document, with different styles in different places (e.g. large title in cover section and repeated in header). It would also be nice to define these attributes in the yaml without having them be automatically included in the document. 

**My Question:** 

I am wondering what the plans are for MS Word support in Quarto. Do you foresee large developments going forward (e.g. implementing some of the features of Officedown, cover pages, support customisation beyond the template file)? Or is this not really a priority? I can understand if this is not a high priority for Quarto, and honestly I would never use Word documents if I could help it. However, given its pervasiveness in industry, I thought I would check if support is coming before I start to hack together a solution.

Any insights here would be greatly appreciated.

Cheers!

**Comment by mcanouil** [comment]
Not really an answer on the "roadmap".

For cover pages, etc., you are likely better off creating a Quarto format extension with custom reference docs document, LUA filters that uses extra YAML parameter, etc.
See for example, <https://nmfs-opensci.github.io/quarto_titlepages/> for cover pages in PDF formats.

See <https://quarto.org/docs/extensions/creating.html> on how to create extensions.
See <https://github.com/kapsner/authors-block> as an example of filter to have a custom author block for Word document.

The main issue with Word is that it is not well supported by Pandoc and is not similar to HTML/PDF (it's already far from perfect between HTML and PDF, not for lack of trying but because they are really different markup languages), _i.e._, Word requires a template file which is a Word document (not a flat file).
The goal of Quarto is to try its best to allow outputs to multiple formats as consistently as it possibly can.
You mentioned `officedown` which was implemented to fill the gap in Pandoc/RMarkdown but it is only for Word/PowerPoint. The author (David Gohel) might want to support Quarto while the team is always open to discuss with maintainers/developers to help get their tool compatible with Quarto (which can also require some changes inside Quarto).

**Reply by mcanouil** [reply]
FYI, the current opened issues regarding Word document support in Pandoc:

- https://github.com/jgm/pandoc/issues?q=is%3Aopen+is%3Aissue+label%3Aformat%3ADocx

**Reply by half-normal** [reply]
Hey @mcanouil , thanks for the reply and for sharing those resources. I really appreciate it!

**Comment by luifrancgom** [comment]
Hello @half-normal. I think this topic is related to pandoc. See  jgm/pandoc#1965 where it is still open. But apparently there are now some implementantations jgm/pandoc#10029 in the case of lof (list of figures) and lot (list of tables) where the Quarto team has this option in mind quarto-dev/quarto-cli#1600
